iron-fish/ironfish
# adds template for migrations on encrypted wallets

## Summary

adds a template migration , 033-encrypted-wallet-template, to demonstrate how to
write wallet migrations on wallet databases that may be encrypted
    
now that account data may be encrypted we need to be careful to decrypt the
account before running the migration and to re-encrypt the migrated data before
writing it back to the database

Closes IFL-3124

## Testing Plan
1. make a copy of an existing datadir, or create a new datadir on the `staging` branch
2. encrypt the wallet from step 1.
3. checkout this branch
4. run `migrations:start`
5. enter passphrase when prompted
6. verify that accounts are still accessible and usable after migration and that wallet is still locked (e.g., run `wallet:accounts`, `wallet:balances`, etc.)

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and label it with `breaking-change-rpc` or `breaking-change-sdk`.

```
[ ] Yes
```
